src/cache.h

```diff
@@ -9,6 +9,14 @@ typedef struct CacheQuery
 {
 	void	   *result;
 	void	   *data;
+	/* The memory context the query was initiated on */
+	MemoryContext mctx;
+
+	/*
+	 * If entry is not found, only allocate a new slot, but do not fill it.
+	 * Defaults to false.
+	 */
+	bool		enteronly;
```

Comment:
Is this better phrased as `fill_new` or something. `enteronly` seems confusing to me. Also is this ever used?

Reply:
Happy to change, but it should probably be something that by default can be false since in this case you need not explicitly specify the default behavior in the scan context.